### PR TITLE
pfTableView: v1.0

### PR DESCRIPTION
**Pull request by dtaylor113**
pfTableView extends angular-datatables, which extends dataTables.net, which is a jQuery plug-in.

v1.0 includes:

- pfTableView as a angular 1.5 component
- row selection via checkbox
- 'Select All' checkbox
- Row action buttons and kabab menus
- Multi-Column Sorting
- Initial pagination controls
- 3 ngDoc examples

Since pfTableView depends on jQuery it is in it's own module.  This allows users to consume pfListView and pfCardVeiw without having to include jQuery.

There are three ngDoc examples for pfTableView:

1. **pfTableView - Basic**: Table view without toolbar.  Just a basic table with row checkbox selection and sorting (mutli-sort by holding down the shift-key), as well as table row action buttons and kabob menu.
2. **pfTableView - With Toolbar**: Table view with toolbar for filtering and bulk actions.  Table view also shows pagination.
3. **pfToolbar**: Updated to include the 'Table' view selector.  Item selection and filtering is persisted when switching between List, Card, and Table views.  Note: the sort dropdown in toolbar is hidden when in table view.

[PTNFLY-2001: pfTableView v1.0](https://patternfly.atlassian.net/browse/PTNFLY-2001)

Follow on JIRA stories:
[PTNFLY-2065: pfTableView v1.5](https://patternfly.atlassian.net/browse/PTNFLY-2065) - Customization of action buttons and kabab menus, Empty State, Unit Tests
[PTNFLY-2083: pfTableView v2.0](https://patternfly.atlassian.net/browse/PTNFLY-2083) - Update to latest designs (pagination, right-aligned action buttons, etc.)
[PTNFLY-1999: pfTableView](https://patternfly.atlassian.net/browse/PTNFLY-1999) - Async. data load

@serenamarie125, @cshinn, @bleathem, @dlabrecq  

Tested in Chrome, IE, and FireFox

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18, @dgutride , @dlabrecq ,  please review :)  -thanks

**Comment by dgutride** [comment]
@jeff-phillips-18  - this looks good to me - are we ready to merge?

**Comment by dlabrecq** [comment]
Wanted to run this locally, but my build appears to fail due to a missing dependency. There is no datatables.net under node_modules, but patternfly has it.

Running "ngdocs:all" (ngdocs) task
Warning: Unable to read "node_modules/datatables.net/js/jquery.dataTables.js" file (Error code: ENOENT). Use --force to continue.
Aborted due to warnings.
Process finished with exit code 3

**Comment by dtaylor113** [comment]
In node_modules I'm showing:
```
datatables.net
datatables.net-dt
datatables.net-select
```

File: src/table/tableview/examples/table-view-basic.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        };
+
+        $scope.colummns = [
+          { header: "Name", itemField: "name" },
+          { header: "Address", itemField: "address"},
+          { header: "City", itemField: "city" },
+          { header: "State", itemField: "state"}
+        ];
+
+        $scope.items = [
```
Comment (dlabrecq) [inline src/table/tableview/examples/table-view-basic.js]: Datatables.net supports different types of data sources. For example:

- HTML (DOM) sourced data
- Ajax sourced data
- Javascript sourced data
- Server-side processing

This example appears to be using the Javascript approach. This is fine, but means the table must be provided with all data upfront (i.e., for pagination to function properly).

Does this implementation also support HTML and/or Ajax sourced data?

**Comment by dlabrecq** [comment]
I now have angular-datatables and datatables.net* under node_modules, but had to edit package.json and move them from optionalDependenies to dependencies.

**Comment by dtaylor113** [comment]
Yes, that is the last follow on story (PTNFLY-1999):

Follow on JIRA stories:
[PTNFLY-2065: pfTableView v1.5](https://patternfly.atlassian.net/browse/PTNFLY-2065) - Customization of action buttons and kabab menus, Empty State, Unit Tests
[PTNFLY-2083: pfTableView v2.0](https://patternfly.atlassian.net/browse/PTNFLY-2083) - Update to latest designs (pagination, right-aligned action buttons, etc.)
[PTNFLY-1999: pfTableView](https://patternfly.atlassian.net/browse/PTNFLY-1999) - Async. data load

File: src/table/tableview/examples/table-view-basic.js (hunk 1 above)
Comment (dtaylor113) [inline src/table/tableview/examples/table-view-basic.js]: Hmm...just did:
```
rm -rf node_modules/
npm install
```
..and I see datatables.net dependcies listed in node_modules.
@jeff-phillips-18 , @dgutride :  can you try these steps and let us know if you get the datatables.net dependencies?  -thanks

**Comment by dlabrecq** [comment]
@dtaylor113 What version of node/npm are you running?

**Comment by dtaylor113** [comment]
angular-patternfly-2>node -v
v6.4.0
angular-patternfly-2>npm -v
3.10.3

**Comment by dlabrecq** [comment]
The packages install fine with npm 3.10.3. I was running 4.0.5, 4.1.2, etc. which had been working fine until now.

**Pull request merged**